subsys/net/l2/ieee802154/ieee802154_mgmt.c

```diff
@@ -260,6 +260,7 @@ static int ieee802154_associate(uint32_t mgmt_request, struct net_if *iface,
 	}
 
 	cmd = ieee802154_get_mac_command(pkt);
+	memset(&cmd->assoc_req.ci, 0, sizeof(cmd->assoc_req.ci));
```

Comment:
just set the 2 last missing fields to 0 (the reserved ones), it will be leaner since all the other fields are set anyway.